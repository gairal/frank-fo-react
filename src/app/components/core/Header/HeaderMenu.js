import React, { Component } from 'react';
import { IconButton, Icon, ListItemIcon, ListItemText, Menu, MenuItem } from '@material-ui/core';

export default class HeaderMenu extends Component {
  constructor(props) {
    super(props);

    this.state = { anchorEl: null };
  }

  handleOpenMenu = (event) => {
    this.setState({ anchorEl: event.currentTarget });
  }

  handleRequestClose = () => {
    this.setState({ anchorEl: null });
  }

  render() {
    return (
      <React.Fragment>
        <IconButton
          color="inherit"
          aria-label="options"
          onClick={this.handleOpenMenu}
        >
          <Icon>more_vert</Icon>
        </IconButton>
        <Menu
          anchorEl={this.state.anchorEl}
          open={Boolean(this.state.anchorEl)}
          onClose={this.handleRequestClose}
        >
          <a
            href="mailto:frank@gairal.com"
            rel="noopener noreferrer"
            title="frank@gairal.com"
          >
            <MenuItem onClick={this.handleRequestClose}>
              <ListItemIcon>
                <Icon>email</Icon>
              </ListItemIcon>
              <ListItemText inset primary="frank@gairal.com" />
            </MenuItem>
          </a>
          <a href="tel:+33695676587" rel="noopener noreferrer" title="+33 6 95 67 65 87">
            <MenuItem onClick={this.handleRequestClose}>
              <ListItemIcon>
                <Icon>phone</Icon>
              </ListItemIcon>
              <ListItemText inset primary="+33 95 67 65 87" />
            </MenuItem>
          </a>
          <a href="https://linkedin.com/in/gairal" target="_blank" rel="noopener noreferrer" title="LinkedIn">
            <MenuItem onClick={this.handleRequestClose}>
              <ListItemIcon>
                <Icon>link</Icon>
              </ListItemIcon>
              <ListItemText inset primary="LinkedIn" />
            </MenuItem>
          </a>
          <a
            href="/pdf/frank.gairal.pdf"
            target="_blank"
            rel="noopener noreferrer"
            title="resume"
          >
            <MenuItem onClick={this.handleRequestClose}>
              <ListItemIcon>
                <Icon>file_download</Icon>
              </ListItemIcon>
              <ListItemText inset primary="resume.pdf" />
            </MenuItem>
          </a>
        </Menu>
      </React.Fragment>
    );
  }
}
