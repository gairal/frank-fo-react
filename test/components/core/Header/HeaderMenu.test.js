/* eslint-disable function-paren-newline,comma-dangle  */
import React from 'react';
import renderer from 'react-test-renderer';
import HeaderMenu from '@/components/core/Header/HeaderMenu';

describe('components > core > Header > HeaderMenu', () => {
  let component;
  beforeEach(() => {
    component = renderer.create(<HeaderMenu />);
  });

  it('Renders correctly default', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
