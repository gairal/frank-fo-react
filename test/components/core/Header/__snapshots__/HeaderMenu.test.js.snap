// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > core > Header > HeaderMenu Renders correctly default 1`] = `
<button
  aria-label="options"
  className="MuiButtonBase-root-7 MuiIconButton-root-1 MuiIconButton-colorInherit-2"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiIconButton-label-6"
  >
    <span
      aria-hidden="true"
      className="material-icons MuiIcon-root-10"
    >
      more_vert
    </span>
  </span>
  <span
    className="MuiTouchRipple-root-16"
  />
</button>
`;
